File: 2279-maximum-bags-with-full-capacity-of-rocks/2279-maximum-bags-with-full-capacity-of-rocks.js
/**
 * @param {number[]} capacity
 * @param {number[]} rocks
 * @param {number} additionalRocks
 * @return {number}
 */
var maximumBags = function(capacity, rocks, additionalRocks) {
  return capacity
    .map((num, i) => num - rocks[i])
    .sort((a, b) => a - b)
    .reduce((acc, curr) => {
      if (curr <= additionalRocks) {
        additionalRocks -= curr;
        acc++;
      }
    
      return acc;
    }, 0)
};

File: 2279-maximum-bags-with-full-capacity-of-rocks/2279-maximum-bags-with-full-capacity-of-rocks.py
class Solution:
    def maximumBags(self, capacity: List[int], rocks: List[int], additionalRocks: int) -> int:
        nums = sorted(c - r for c, r in zip(capacity, rocks))
        ans = 0
        
        for num in nums:
            if additionalRocks >= num:
                additionalRocks -= num
                ans += 1
            else:
                break
        return ans